examples/medplum-chart-demo/src/components/note-display/EncounterNoteDisplay.tsx

```diff
@@ -0,0 +1,98 @@
+import { Group, Stack, Title } from '@mantine/core';
+import { formatDate, getQuestionnaireAnswers } from '@medplum/core';
+import { Encounter, Questionnaire, QuestionnaireResponse } from '@medplum/fhirtypes';
+import { CodeableConceptDisplay, Document, QuantityDisplay, useMedplum } from '@medplum/react';
+import { parseGenericAnswers } from './utils';
+import { GeneralNoteDisplay } from './GeneralNoteDisplay';
+import { GynecologyNoteDisplay } from './GynecologyNoteDisplay';
+import { ObstetricNoteDisplay } from './ObstetricNoteDisplay';
+
+interface EncounterNoteDisplayProps {
+  response: QuestionnaireResponse;
+  encounter: Encounter;
+}
+
+export function EncounterNoteDisplay(props: EncounterNoteDisplayProps): JSX.Element {
+  const medplum = useMedplum();
+
+  // Ensure that the correct response is being displayed
+  function checkForValidResponse(): void {
+    const response = props.response;
+    const encounter = props.encounter;
+
+    if (response.encounter?.reference !== `Encounter/${encounter.id}`) {
+      throw new Error('Invalid note');
+    }
+  }
+  checkForValidResponse();
+
+  function getNoteType(): string {
+    const response = props.response;
+    // Get the questionnaire the response is linked to
+    const questionnaire = medplum.readReference({ reference: response.questionnaire }).read() as Questionnaire;
+
+    // If there is no use context, return the default note type
+    if (!questionnaire.useContext) {
+      return 'general';
+    }
+
+    // Check the use contexts. In this demo, we have a gynecology, obstetric, and general note type
+    for (const use of questionnaire.useContext) {
+      // Check for gynecology
+      if (use.valueCodeableConcept?.coding?.[0].code === '83607001') {
+        return 'gynecology';
```

Comment:
Instead of converting the code to a string, can't you just return the code, and use that code to search for the questionnaire?

Reply:
Grab the use context with `code` equal to `focus`. That way you don't have to do this kind of loop with early exits